test/unittests/execute_call_test.cpp

```diff
@@ -533,8 +535,54 @@ TEST(execute_call, call_max_depth)
     const auto module = parse(bin);
     auto instance = instantiate(module);
 
-    EXPECT_THAT(execute(*instance, 0, {}, MaxDepth), Result(42));
-    EXPECT_THAT(execute(*instance, 1, {}, MaxDepth), Traps());
+    EXPECT_THAT(execute(*instance, 0, {}, MaxDepth - 1), Result(42));
+    EXPECT_THAT(execute(*instance, 1, {}, MaxDepth - 1), Traps());
+}
+
+TEST(execute_call, call_initial_depth)
+{
+    /* wat2wasm
+    (import "mod" "foo" (func))
+    */
+
+    const auto wasm = from_hex("0061736d01000000010401600000020b01036d6f6403666f6f0000");
+
+    const auto module = parse(wasm);
+    auto host_foo = [](Instance& /*instance*/, span<const Value>, int depth) -> ExecutionResult {
+        EXPECT_EQ(depth, 0);
+        return Void;
+    };
+    const auto host_foo_type = module.typesec[0];
+
+    auto instance = instantiate(module, {{host_foo, host_foo_type}});
+
+    EXPECT_THAT(execute(*instance, 0, {}), Result());
+}
+
+TEST(execute_call, execute_imported_max_depth)
+{
+    /* wat2wasm
+    (import "mod" "foo" (func))
+    (func)
+    */
+
+    const auto wasm =
+        from_hex("0061736d01000000010401600000020b01036d6f6403666f6f0000030201000a040102000b");
+
+    const auto module = parse(wasm);
+    auto host_foo = [](Instance& /*instance*/, span<const Value>, int depth) -> ExecutionResult {
+        if (depth + 1 > MaxDepth)
```

Comment:
I would argue that this is logical thing for a host function to do, because that's what `execute` does

Reply:
But `execute` does this check prior to the host function is executed. Or I'm missing something in the mess we have.

Reply:
Alright, true, this is not needed, disregard this test for now.